tests/test_helpers.py

```diff
@@ -34,240 +34,230 @@ def has_encoding(name):
-        c = app.test_client()
-        rv = c.post('/json', data='"foo"', content_type='application/x+json')
+
+        rv = client.post('/json', data='"foo"', content_type='application/x+json')
         assert rv.data == b'foo'
 
-    def test_json_body_encoding(self):
-        app = flask.Flask(__name__)
+    def test_json_body_encoding(self, app, client):
         app.testing = True
+
         @app.route('/')
         def index():
             return flask.request.get_json()
 
-        c = app.test_client()
-        resp = c.get('/', data=u'"Hällo Wörld"'.encode('iso-8859-15'),
-                     content_type='application/json; charset=iso-8859-15')
+        resp = client.get('/', data=u'"Hällo Wörld"'.encode('iso-8859-15'),
+                          content_type='application/json; charset=iso-8859-15')
         assert resp.data == u'Hällo Wörld'.encode('utf-8')
 
-    def test_json_as_unicode(self):
-        app = flask.Flask(__name__)
+    def test_json_as_unicode(self, app, app_ctx):
 
         app.config['JSON_AS_ASCII'] = True
-        with app.app_context():
-            rv = flask.json.dumps(u'\N{SNOWMAN}')
-            assert rv == '"\\u2603"'
+        rv = flask.json.dumps(u'\N{SNOWMAN}')
+        assert rv == '"\\u2603"'
 
         app.config['JSON_AS_ASCII'] = False
-        with app.app_context():
-            rv = flask.json.dumps(u'\N{SNOWMAN}')
-            assert rv == u'"\u2603"'
+        rv = flask.json.dumps(u'\N{SNOWMAN}')
+        assert rv == u'"\u2603"'
 
-    def test_json_dump_to_file(self):
-        app = flask.Flask(__name__)
+    def test_json_dump_to_file(self, app, app_ctx):
         test_data = {'name': 'Flask'}
         out = StringIO()
 
-        with app.app_context():
-            flask.json.dump(test_data, out)
-            out.seek(0)
-            rv = flask.json.load(out)
-            assert rv == test_data
+        flask.json.dump(test_data, out)
+        out.seek(0)
+        rv = flask.json.load(out)
+        assert rv == test_data
 
     @pytest.mark.parametrize('test_value', [0, -1, 1, 23, 3.14, 's', "longer string", True, False, None])
-    def test_jsonify_basic_types(self, test_value):
+    def test_jsonify_basic_types(self, test_value, app, client):
         """Test jsonify with basic types."""
-        app = flask.Flask(__name__)
-        c = app.test_client()
 
         url = '/jsonify_basic_types'
         app.add_url_rule(url, url, lambda x=test_value: flask.jsonify(x))
-        rv = c.get(url)
+        rv = client.get(url)
         assert rv.mimetype == 'application/json'
         assert flask.json.loads(rv.data) == test_value
 
-    def test_jsonify_dicts(self):
+    def test_jsonify_dicts(self, app, client):
         """Test jsonify with dicts and kwargs unpacking."""
         d = dict(
             a=0, b=23, c=3.14, d='t', e='Hi', f=True, g=False,
             h=['test list', 10, False],
-            i={'test':'dict'}
+            i={'test': 'dict'}
```

Comment:
What do you think about switching this from the `dict(key=value)` format to the more common `{key: value}` format? It's not a requirement at all, just a thought...